docs/examples/caffe_build_index.md

```diff
@@ -0,0 +1,54 @@
+The following is a concrete example of performing nearest neiborhood computation
+using a set of ten butterfly images. It assumes you have set up the
+caffe (vrc2) environment (http://caffe.berkeleyvision.org/install_apt.html)
```

Comment:
when I tried to use master of caffe, I ran into a build issue but I will update here saying that it should work with caffe master as well.